client/src/components/DetailsEvent/index.js

```diff
@@ -1,13 +1,180 @@
-import React from 'react'
-import Header from '../Header/header';
-class DetailsEvent extends React.Component {
+import React, { Component } from "react";
+import NavBar from "../NavBar";
+import SideBar from "../SideBar";
+import axios from "axios";
+
+class DetailsEvent extends Component {
+  state = {
+    id: this.props.history.location.event.id,
+    status: this.props.history.location.event.status,
+    loading: false
+  };
+
+  componentDidMount() {
+    axios
+      .get(`/event/${this.state.id}`)
+      .then(data => {
+        this.setState({
+          event_details: data.data.event,
+          coffee_details: data.data.coffee,
+          lunch_details: data.data.lunch,
+          equipment_details: data.data.equipment,
+          loading: true
+        });
+      })
+      .catch(() => {
+        const { history } = this.props;
+        history.push("/error");
+      });
+  }
+
+  handleSubmitForm = event => {
```

Comment:
Why not just make multiple submit functions, and pass the correct one depending on the status check you are doing down on line 166?

Like, `handleApprove`, `handleCancel` or something?